src/NHibernate/SqlCommand/SqlSelectBuilder.cs

```diff
@@ -40,6 +42,7 @@ public SqlSelectBuilder SetComment(string comment)
 		public SqlSelectBuilder SetFromClause(string fromClause)
 		{
 			this.fromClause = fromClause;
+			hasOuterJoin = hasOuterJoin || fromClause.ToLowerInvariant().Contains("left outer join");
```

Comment:
Better use [this code](https://github.com/nhibernate/nhibernate-core/pull/1567/files#diff-92318c5b35fd56f9c1354dca5d86c4bbR693) for a case insensitive `Contains`. It has been merge just now, so update your branch for having this `StringHelper.ContainsCaseInsensitive` method.

The test should be moved into `GetForUpdateString`, and done only if the dialect does not support for update on outer joins. It would avoid doing such a string scan for all other cases.

Reply:
Changed to `StringHelper.ContainsCaseInsensitive`; and moved the check for outer join into `GetForUpdateString()`.